state-chain/custom-rpc/src/lib.rs

```diff
@@ -1347,16 +1354,32 @@ where
 		&self,
 		at: Option<state_chain_runtime::Hash>,
 	) -> RpcResult<Vec<scale_value::Value>> {
-		let event_decoder = type_decoder::TypeDecoder::new::<state_chain_runtime::RuntimeEvent>();
+		let event_decoder = type_decoder::TypeDecoder::new::<
+			frame_system::EventRecord<state_chain_runtime::RuntimeEvent, state_chain_runtime::Hash>,
+		>();
 		let events = self
 			.client
 			.runtime_api()
-			.cf_get_events(self.unwrap_or_best(at))
+			.cf_get_events(self.unwrap_or_best(at), EventFilter::AllEvents)
 			.map_err(to_rpc_error)?;
 
 		Ok(events
 			.into_iter()
-			.map(|event| event_decoder.decode_data(event))
+			.map(|event_record| event_decoder.decode_data(event_record.encode()))
+			.collect::<Vec<_>>())
+	}
+
+	fn cf_get_system_events_encoded(
+		&self,
+		at: Option<state_chain_runtime::Hash>,
+	) -> RpcResult<Vec<String>> {
```

Comment:
Instead of returning a string we should use `sp_core::Bytes` for example, and rely on the serializer to do the hex encoding.